### PR TITLE
Core events documentation

### PR DESCRIPTION
**Pull request by sideninja**
As discussed moved from flow repo: https://github.com/onflow/flow/pull/591

## Description
Documentation of core contract and native events emitted on Flow. There were many cases where developers needed to track most basic events but struggled with event names or their meaning, this document tries to solve that.

This work is part of the sprint tasks to improve the documentation.

______

- [x] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: docs/core-events.mdx [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+
+### Account Key Added
+
+Event that is emitted when a key gets added to an account.
+
+Event name: `flow.AccountKeyAdded`
+
+```cadence
+pub event AccountKeyAdded(address: Address)
```
Comment (joshuahannan) [inline docs/core-events.mdx:65]: looks like you didn't include the publickey parameter here

File: docs/core-events.mdx [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+
+
+### Account Key Removed
+
+Event that is emitted when a key gets removed from an account.
+
+Event name: `flow.AccountKeyRemoved`
+
+```cadence
+pub event AccountKeyRemoved(address: Address)
```
Comment (joshuahannan) [inline docs/core-events.mdx:81]: didn't include the publicKey parameter here

File: docs/core-events.mdx [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+- Testnet event: `A.7e60df042a9c0868.FlowToken.TokensWithdrawn`
+
+```cadence
+pub event TokensWithdrawn(amount: UFix64, from: Address?)
+```
+
+| Field             | Type   | Description                                                            |
+| ----------------- | ------ | ---------------------------------------------------------------------- |
+| amount       | UFix64 | The amount of tokens withdrawn |
+| from       | Address? | Optional address of the account to withdraw from |
```
Comment (joshuahannan) [inline docs/core-events.mdx:194]:
```suggestion
| from       | Address? | Optional address of the account that owns the vault where tokens were withdrawn from. `nil` if the vault is not in an account's storage
```

File: docs/core-events.mdx [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+- Testnet event: `A.7e60df042a9c0868.FlowToken.TokensDeposited`
+
+```cadence
+pub event TokensDeposited(amount: UFix64, to: Address?)
+```
+
+| Field             | Type   | Description                                                            |
+| ----------------- | ------ | ---------------------------------------------------------------------- |
+| amount       | UFix64 | The amount of tokens withdrawn |
+| to       | Address? | Optional address of the account the amount is deposited to |
```
Comment (joshuahannan) [inline docs/core-events.mdx:212]:
```suggestion
| to       | Address? | Optional address of the account that owns the vault where tokens were deposited to. `nil` if the vault is not in an account's storage |
```

**Review by joshuahannan — APPROVED** [review]
looks good! Just a few comments

**Comment by turbolent** [comment]
Thank you for documenting this @sideninja!

The `/docs` directory only contains developer documentation for the  Cadence implementation, i.e. has documentation for contributors, not users. I would suggest moving the document to the `/docs/language` directory, so it can be shown on docs.onflow.org (once integrated there).

Also, I think the documentation for Cadence should only include the core built-in events (account creation, contract deployment, etc.) the `FlowToken` events are not Cadence events and should probably be documented here instead: https://docs.onflow.org/core-contracts/flow-token/

I would also move the instructions for how to query the events using SDKs/CLI elsewere, as we only keep Cadence documentation in this repo.

**Comment by sideninja** [comment]
> Thank you for documenting this @sideninja!
> 
> The `/docs` directory only contains developer documentation for the Cadence implementation, i.e. has documentation for contributors, not users. I would suggest moving the document to the `/docs/language` directory, so it can be shown on https://docs.onflow.org (once integrated there). Maybe `built-in-events.md`?
> 
> Also, I think the documentation for Cadence should only include the core built-in events (account creation, contract deployment, etc.) the `FlowToken` events are not Cadence events and should probably be documented here instead: https://docs.onflow.org/core-contracts/flow-token/
> 
> I would also move the instructions for how to query the events using SDKs/CLI elsewere, as we only keep Cadence documentation in this repo.

Yeah, I agree, this was moved from another location and it's very awkward to place it, but I 100% agree on not polluting Cadence docs with things that are not related to Cadence specifically. Let me try to fix this.

**Comment by sideninja** [comment]
> > Thank you for documenting this @sideninja!
> > The `/docs` directory only contains developer documentation for the Cadence implementation, i.e. has documentation for contributors, not users. I would suggest moving the document to the `/docs/language` directory, so it can be shown on https://docs.onflow.org (once integrated there). Maybe `built-in-events.md`?
> > Also, I think the documentation for Cadence should only include the core built-in events (account creation, contract deployment, etc.) the `FlowToken` events are not Cadence events and should probably be documented here instead: https://docs.onflow.org/core-contracts/flow-token/
> > I would also move the instructions for how to query the events using SDKs/CLI elsewere, as we only keep Cadence documentation in this repo.
> 
> Yeah, I agree, this was moved from another location and it's very awkward to place it, but I 100% agree on not polluting Cadence docs with things that are not related to Cadence specifically. Let me try to fix this.

@turbolent are you ok with this being merged?

**Review by turbolent — APPROVED** [review]
Looks good! Next step would be to add this to the sidebar in the `onflow/flow` repo

**Comment by sideninja** [comment]
> Looks good! Next step would be to add this to the sidebar in the `onflow/flow` repo

cc @10thfloor can you help

**Pull request merged**